### DISCUSSION TITLE
about the Batch request length too long

### DISCUSSION THREAD

**Discussion by jason6165808** [blockscout/blockscout#7025] [Support]
Hey,Can you help me? How to fix it? thank you.

2023-03-06T16:41:17.844 [error] Task #PID<0.14154.0> started from #PID<0.14140.0> terminating
** (FunctionClauseError) no function clause matching in EthereumJSONRPC.HTTP.standardize_response/1
    (ethereum_jsonrpc 5.1.1) lib/ethereum_jsonrpc/http.ex:145: EthereumJSONRPC.HTTP.standardize_response(%{"error" => %{"code" => -32600, "message" => "Batch request length too long"}, "jsonrpc" => "2.0"})
    (elixir 1.13.4) lib/enum.ex:1593: Enum."-map/2-lists^map/1-0-"/2
    (ethereum_jsonrpc 5.1.1) lib/ethereum_jsonrpc/http.ex:44: EthereumJSONRPC.HTTP.chunked_json_rpc/3
    (ethereum_jsonrpc 5.1.1) lib/ethereum_jsonrpc/request_coordinator.ex:86: anonymous fn/3 in EthereumJSONRPC.RequestCoordinator.perform/4
    (ethereum_jsonrpc 5.1.1) lib/ethereum_jsonrpc/request_coordinator.ex:106: EthereumJSONRPC.RequestCoordinator.trace_request/2
    (ethereum_jsonrpc 5.1.1) lib/ethereum_jsonrpc/receipts.ex:136: EthereumJSONRPC.Receipts.fetch/2
    (elixir 1.13.4) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.13.4) lib/task/supervised.ex:34: Task.Supervised.reply/4
Function: &:erlang.apply/2
    Args: [#Function<0.124558711/1 in Indexer.Block.Fetcher.Receipts.fetch/2>, [[%{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0x57f1D85dEF052D2228fFD2A9aFa8A9C6bA764459", gas: 21000, gas_price: 10000, hash: "0x2c994e51070958aaa59b06491b3dd9d8d129b8b576ddf3d96a1a18e0f73db9dc", index: 0, input: "0x", nonce: 19, r: 107755194116052800307311255567849561820500315152206058247669499782753107609682, s: 924472046626493177577307511585239934027236983023462682254859646043394671514, to_address_hash: "0x57f1D85dEF052D2228fFD2A9aFa8A9C6bA764459", transaction_index: 0, v: 28, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0xe6A8DcaA99b9B3D66C5838D727A5082Fad43D714", gas: 21000, gas_price: 10000, hash: "0x4dc6c77042e44fcb4ac1a181b4430e4911a0d8d81fce467019f3be11f3a9dae0", index: 1, input: "0x", nonce: 14, r: 31899195454309522383663440437400602427494364225142696754341277399695925856483, s: 15942354207243025829878286811740674066851121950734960465261077365990955519027, to_address_hash: "0xe6A8DcaA99b9B3D66C5838D727A5082Fad43D714", transaction_index: 1, v: 28, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0xc36e33833D58138FB42Bd947206C405ca9EC265f", gas: 21000, gas_price: 10000, hash: "0x4775148ae8509b07fe9e43de1c7f01506b3c0633d5dc69596f4a585319ff5575", index: 2, input: "0x", nonce: 35, r: 65880350499865734499254155092216877225682014343633803584155796577066250002974, s: 52938158171481944842488294598271603679896958775260462936470159908329405738032, to_address_hash: "0xc36e33833D58138FB42Bd947206C405ca9EC265f", transaction_index: 2, v: 27, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0x6EF08ECEBAcb5520c3C0cD8cC955B34A5B716A06", gas: 21000, gas_price: 10000, hash: "0xdc7ff8d98214ed67852b02c4ac54f0b97d2f8f2dddf2fd86c2b5782ce7392b31", index: 3, input: "0x", nonce: 26, r: 113944483570208324520470157230077712582300249855622416747464565661841189871001, s: 27264181179658908994299389010854377900112170761751072216616975026313443682796, to_address_hash: "0x6EF08ECEBAcb5520c3C0cD8cC955B34A5B716A06", transaction_index: 3, v: 28, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0xfAdaA05e28e95Fecab3D2C3e3509d1212615F0e8", gas: 21000, gas_price: 10000, hash: "0xe7769d4537a6a2c75d7ef44492cd60e68cb186f37d924d578f9eb3312355de3f", index: 4, input: "0x", nonce: 14, r: 41409949356946079159367818870108874234994797975316585989007168913424073619419, s: 20563149817265341796445236253139261139281438971151175151918683295185158569055, to_address_hash: "0xfAdaA05e28e95Fecab3D2C3e3509d1212615F0e8", transaction_index: 4, v: 27, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0x53386E337D8D89bf99868BF7c8Fc391b5D096991", gas: 21000, gas_price: 10000, hash: "0x798b1c99cf7389bf3b42496db094b5d1a45f044849ea7503dc411c3f03da565b", index: 5, input: "0x", nonce: 4, r: 106094793778721877188071519019619341165089109440421272470392922179754420107761, s: 39863662763915593052154970326525699853208243401118663109765469028737008871976, to_address_hash: "0x53386E337D8D89bf99868BF7c8Fc391b5D096991", transaction_index: 5, v: 28, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0xC8A3B2C8FF612D02B586F7E7BE0d614050818C57", gas: 21000, gas_price: 10000, hash: "0x1bf0e4938cb7701f7f6e65aa0151163e9971b52ca20123bfb92d18027befd7fe", index: 6, input: "0x", nonce: 13, r: 5470062918815423541938846527527180974943797401520722988860877588132356244772, s: 10506353522467197140538379743122128860811922137698120963835218256348120468882, to_address_hash: "0xC8A3B2C8FF612D02B586F7E7BE0d614050818C57", transaction_index: 6, v: 27, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0xe458248C4263bf5ed41A755364b7B86C67742FE1", gas: 21000, gas_price: 10000, hash: "0xa10916c2939a73f1500b1522f2f4cd864bdc4ae087f91f67bc28f3b66a54fb1c", index: 7, input: "0x", nonce: 8, r: 76715411829026273187053440423669676277942046840128159971305356520911269108567, s: 18123363819110275956194706382669875177031271024300226854486597785051960515250, to_address_hash: "0xe458248C4263bf5ed41A755364b7B86C67742FE1", transaction_index: 7, v: 27, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0xFAbEff4389637B91E5de94b94f0de12542E0EaCB", gas: 21000, gas_price: 10000, hash: "0x33e6e48ee6691d6464ddc919dff5672fcc6ce226c032205efb0eda4de653855a", index: 8, input: "0x", nonce: 36, r: 81830467221056084116863451728093200803264682715444169095723978302062432171476, s: 35707357474749657637737739880913483500337242577622023115421660916732074524588, to_address_hash: "0xFAbEff4389637B91E5de94b94f0de12542E0EaCB", transaction_index: 8, v: 28, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0xF1663dc250c54e13287A8b2752631A4156ab89d3", gas: 21000, gas_price: 10000, hash: "0xe80e8f9284c34797c038de8e51a8097d87fc8ced620c58fe21043ca95180a783", index: 9, input: "0x", nonce: 30, r: 37817565099239205694297311045386778074392368685758083420728236783057485634914, s: 10746918245131561917158262324050899750049421206285457801847665355852697101323, to_address_hash: "0xF1663dc250c54e13287A8b2752631A4156ab89d3", transaction_index: 9, v: 28, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0x838cd6C730DEC9343FBD3f96873C04799B19CFDd", gas: 21000, gas_price: 10000, hash: "0x24d5a4be865a82f44468540fef77bc17627e31cc765f0e28816a89b78496acaf", index: 10, input: "0x", nonce: 8, r: 88870495212137676389583460219684224908064361337265068791865042816607903970904, s: 18674745638747995982587359721390494576552045653567482326843678182956120341724, to_address_hash: "0x838cd6C730DEC9343FBD3f96873C04799B19CFDd", transaction_index: 10, v: 28, value: 1}, %{block_hash: "0x952e6fb2728b96c4a6d7aa5861b9dbe53808446abdc2d24a071d8dce823bd31f", block_number: 117315, from_address_hash: "0x446cA6c8eDe854bFd759D06726d4D97286B8b4C6", gas: 21000, gas_price: 10000, hash: "0x9ac869acca3a26c5c1040109c007b5ff1f4db5e433e7f7808c2f424f359df65c", index: 11, input: "0x", nonce: 6, r: 1388943437757829686047114263037652195127055322323020519995622898627004225753, s: 35268980484275857203658370958866978101847142629664284202817490934530082175974, to_address_hash: "0x446cA6c8eDe854bFd759D06726d4D97286B8b4C6", transaction_index: 11, v: 27, value: 1} (truncated)
2023-03-06T16:41:17.853 application=indexer fetcher=block_catchup first_block_number=117324 last_block_number=117315 [error] ** (FunctionClauseError) no function clause matching in anonymous fn/2 in Indexer.Block.Fetcher.Receipts.fetch/2
    (indexer 5.1.1) anonymous fn({:exit, {%FunctionClauseError{args: nil, arity: 1, clauses: nil, function: :standardize_response, kind: nil, module: EthereumJSONRPC.HTTP}, [{EthereumJSONRPC.HTTP, :standardize_response, [%{"error" => %{"code" => -32600, "message" => "Batch request length too long"}, "jsonrpc" => "2.0"}], [file: 'lib/ethereum_jsonrpc/http.ex', line: 145]}, {Enum, :"-map/2-lists^map/1-0-", 2, [file: 'lib/enum.ex', line: 1593]}, {EthereumJSONRPC.HTTP, :chunked_json_rpc, 3, [file: 'lib/ethereum_jsonrpc/http.ex', line: 44]}, {EthereumJSONRPC.RequestCoordinator, :"-perform/4-fun-0-", 3, [file: 'lib/ethereum_jsonrpc/request_coordinator.ex', line: 86]}, {EthereumJSONRPC.RequestCoordinator, :trace_request, 2, [file: 'lib/ethereum_jsonrpc/request_coordinator.ex', line: 106]}, {EthereumJSONRPC.Receipts, :fetch, 2, [file: 'lib/ethereum_jsonrpc/receipts.ex', line: 136]}, {Task.Supervised, :invoke_mfa, 2, [file: 'lib/task/supervised.ex', line: 89]}, {Task.Supervised, :reply, 4, [file: 'lib/task/supervised.ex', line: 34]}]}}, {:ok, %{logs: [], receipts: []}}) in Indexer.Block.Fetcher.Receipts.fetch/2
    (elixir 1.13.4) lib/task/supervised.ex:367: Task.Supervised.stream_deliver/7
    (elixir 1.13.4) lib/enum.ex:2442: Enum.reduce_while/3
    (indexer 5.1.1) lib/indexer/block/fetcher/receipts.ex:22: Indexer.Block.Fetcher.Receipts.fetch/2
    (indexer 5.1.1) lib/indexer/block/fetcher.ex:137: Indexer.Block.Fetcher.fetch_and_import_range/2
    (stdlib 3.16) timer.erl:166: :timer.tc/1
    (indexer 5.1.1) lib/indexer/block/catchup/fetcher.ex:180: Indexer.Block.Catchup.Fetcher.fetch_and_import_range_from_sequence/3
    (elixir 1.13.4) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2

**Comment by varasev** [comment]
You can try to set `INDEXER_RECEIPTS_BATCH_SIZE` env variable explicitly. By default its value is `250`. Try to set it to some smaller value
